### ISSUE TITLE
md_document(variant="markdown", preserve_yaml=T) produces two YAML blocks

### ISSUE THREAD

**Issue by agoldst** [rstudio/rmarkdown#656]
The `md_document` format interacts strangely with the `preserve_yaml` option when the specified markdown variant is `markdown` (ie pandoc markdown).

Consider the following `test.Rmd`:

``````

---
title: Example
output:
  md_document:
    variant: markdown
    preserve_yaml: true

---

An example.

```{r}
2 + 2
```
``````

Converting this with `rmarkdown::render("test.Rmd")` produces the following on my Mac with either rmarkdown from CRAN or the latest version from github:

``````

---
title: Example
output:
  md_document:
    variant: markdown
    preserve_yaml: true

---

---
output:
  md_document:
    preserve_yaml: True
    variant: markdown
title: Example
...

An example.

``` {.r}
2 + 2
```

    ## [1] 4
``````

This appears to be because `md_document` always passes pandoc the `--standalone` option (see [here](https://github.com/rstudio/rmarkdown/blob/master/R/md_document.R#L63)), which causes pandoc to output the yaml metadata itself (converting the terminating delimiter to `...`), appended after the metadata block from rmarkdown's output. This does not happen with the default `markdown_strict` variant, only `markdown`. May I suggest not giving pandoc the `--standalone` option when `preserve_yaml` is true?

The creates problems when knitting Rmd to markdown with metadata for static-site generators like jekyll.

**Comment by yihui** [comment]
Since you were trying to build websites based on R Markdown, I recommend you to take a look at blogdown: https://github.com/rstudio/blogdown. It primarily supports Hugo, but you can use Jekyll, too.

Anyway, this issue sounds like a bug to me. Thanks for the report!

**Comment by agoldst** [comment]
Thanks for the response and for all your marvelous work on literate programming with R. I threw together my own hugo + rmarkdown setup before you released blogdown...Some day I may try to switch over.

**Comment by yihui** [comment]
I sent a fix at #1170. You may test it with `devtools::install_github('rstudio/rmarkdown#1170')`.
